core/src/main/java/com/linecorp/armeria/server/http/healthcheck/ManagedHttpHealthCheckService.java

```diff
@@ -0,0 +1,110 @@
+
+import com.google.common.base.Ascii;
+
+import com.linecorp.armeria.common.http.AggregatedHttpMessage;
+import com.linecorp.armeria.common.http.HttpData;
+import com.linecorp.armeria.common.http.HttpRequest;
+import com.linecorp.armeria.common.http.HttpResponseWriter;
+import com.linecorp.armeria.common.http.HttpStatus;
+import com.linecorp.armeria.server.ServiceRequestContext;
+
+/**
+ * A {@link HttpHealthCheckService} which allows overriding its status via a POST request.
+ * <p>
+ * <h2>Example</h2>
+ * <pre>{@code
+ * Server server = new ServerBuilder().serviceUnder("/health", new ManagedHttpHealthCheckService())
+ *                                    .build()
+ * }</pre>
+ * Default config uses content on or off
+ * You can also use your own path matching.
+ * </p>
+ * <pre>{@code
+ * ...
+ * Server server = new ServerBuilder().serviceUnder("health", new ManagedHttpHealthCheckService() {
+ *                         public CompletableFuture<Optional<Boolean>> mode(HttpRequest req) {
+ *                                 return CompletableFuture.completedFuture(Optional.empty());
+ *                         }
+ *                 }).build())
+ * ...
+ * }</pre>
+ */
+public class ManagedHttpHealthCheckService extends HttpHealthCheckService {
+    private static final AggregatedHttpMessage TURN_ON_RES = AggregatedHttpMessage
+            .of(HttpStatus.OK, HttpData.ofAscii("turn on"));
+    private static final AggregatedHttpMessage TURN_OFF_RES = AggregatedHttpMessage
+            .of(HttpStatus.OK, HttpData.ofAscii("turn off"));
+    private static final AggregatedHttpMessage INTERNAL_SERVER_ERROR_RES = AggregatedHttpMessage
+            .of(HttpStatus.INTERNAL_SERVER_ERROR);
+
+    @Override
+    protected void doPut(ServiceRequestContext ctx, HttpRequest req, HttpResponseWriter res) throws Exception {
+        updateHealthStatus(ctx, req).thenAccept(res::respond)
+                                    .exceptionally(t -> {
+                                        res.respond(INTERNAL_SERVER_ERROR_RES);
+                                        return null;
+                                    });
+    }
+
+    /**
+     * Updates health status using the specified {@link HttpRequest}.
+     */
+    private CompletableFuture<AggregatedHttpMessage> updateHealthStatus(
+            ServiceRequestContext ctx, HttpRequest req) {
+        return mode(req).thenApply(mode -> {
+            if (!mode.isPresent()) {
+                return newResponse(ctx);
+            }
+
+            boolean tobe = mode.get();
+
+            serverHealth.setHealthy(tobe);
+
+            return tobe ? TURN_ON_RES : TURN_OFF_RES;
+        });
+    }
+
+    /**
+     * Judge the turning mode.
+     * True means turn on, False is turn off.
+     * If not present, Not supported.
+     * Default implementation is check content is on/off for PUT method
+     *
+     * @param req HttpRequest
+     */
+    protected CompletableFuture<Optional<Boolean>> mode(HttpRequest req) {
```

Comment:
Could you also add `ServiceRequestContext` as the first parameter of this method?